layouts/_default/search.html

```diff
@@ -3,57 +3,189 @@
+
+        <label for="lang-filter">Language:</label>
+        <select id="lang-filter">
+            {{ range $languages }}
+                <option value="{{ .code }}">{{ .name }}</option>
             {{ end }}
-            if(window.location.href.includes('&site=')){
-                cx = '{{ .Site.Data.args.docs_search_engine_id }}';
-            }
-                const gcse = document.createElement('script');
-                gcse.type = 'text/javascript';
-                gcse.async = true;
-                gcse.src = 'https://cse.google.com/cse.js?cx=' + cx;
-                const s = document.getElementsByTagName('script')[0];
-                s.parentNode.insertBefore(gcse, s);
-            })();
-    </script>
-
-    {{ if .Site.Data.args.archive }}
-        <gcse:searchresults-only defaultToRefinement="{{ .Site.Data.args.archive_search_refinement }}"></gcse:searchresults-only>
-    {{ else }}
-        <gcse:searchresults-only></gcse:searchresults-only>
-    {{ end }}
+        </select>
+
+        <label for="version-filter">Version:</label>
+        <select id="version-filter">
+            {{ range $versions }}
+                <option value="{{ . }}">{{ . }}</option>
+            {{ end }}
+        </select>
+    </div>
+    <div id="search-query-header"></div>
+    <div id="search-results">Loading results...</div>
+    <div id="pagination-controls">
+        <div><button id="prev-page" class="btn" disabled>Previous</button></div>
+        <span id="page-info"></span>
+        <div><button id="next-page" class="btn" disabled>Next</button></div>
+    </div>
 </div>
 
+<script src="https://cdn.jsdelivr.net/npm/algoliasearch@4/dist/algoliasearch-lite.umd.js"></script>
 <script>
-    function getParameterByName(name, url) {
-        if (!url) {
-            url = window.location.href;
-        }
+    const appId = '95IG3UJ1LV';
+    const apiKey = '23827432dd1f4c529eece856289a421c';
+    const indexName = 'istio';
+
+    const searchClient = algoliasearch(appId, apiKey);
+    const index = searchClient.initIndex(indexName);
+
+    function getParameterByName(name, url = window.location.href) {
         name = name.replace(/[\[\]]/g, "\\$&");
-        const regex = new RegExp("[?&]" + name + "(=([^&#]*)|&|#|$)"),
-                results = regex.exec(url);
-        if (!results) {
-            return null;
-        }
-        if (!results[2]) {
-            return '';
-        }
+        const regex = new RegExp("[?&]" + name + "(=([^&#]*)|&|#|$)");
+        const results = regex.exec(url);
+        if (!results) return null;
+        if (!results[2]) return '';
         return decodeURIComponent(results[2].replace(/\+/g, " "));
     }
-    const q = getParameterByName('q', window.location.href);
-    document.getElementsByName('q')[0].value = q;
+
+    function extractVersionAndLangFromPath() {
+        const pathParts = window.location.pathname.split('/');
+        let version = 'latest';
+        let lang = 'en';
```

Comment:
Did you write this code, or is it based on something from Algolia?
Should it be tested?  Should it be in its own JS/TS file?